code/modules/power/lgenerator.dm

```diff
@@ -11,355 +11,349 @@ TYPEINFO(/obj/machinery/power/lgenerator)
-		light.attach(src)
-		light.set_brightness(0.8)
-		src.spin_sprite = new /image(src.icon,"ggen-spin")
-		src.tank_sprite = new /image(src.icon,"ggen-tank")
-
-	attackby(obj/item/W, mob/user)
-		if (istype(W, /obj/item/tank/))
-			if (src.P)
-				user.show_text("There appears to be a tank loaded already.", "red")
-				return
-			if (src.check_tank(W) == 0)
-				user.show_text("The tank doesn't contain any plasma.", "red")
-				return
-			src.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>")
-			user.u_equip(W)
-			W.set_loc(src)
-			src.P = W
-			src.UpdateIcon()
-
-		else if (istype(W, /obj/item/cell))
-			if (src.CL)
-				user.show_text("There appears to be a power cell inserted already.", "red")
-				return
-			src.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>")
-			user.u_equip(W)
-			W.set_loc(src)
-			src.CL = W
-
-		else
-			..()
+/obj/machinery/power/lgenerator/New()
+	..()
+	light = new /datum/light/point
+	light.attach(src)
+	light.set_brightness(0.8)
+	src.spin_sprite = new /image(src.icon,"ggen-spin")
+	src.tank_sprite = new /image(src.icon,"ggen-tank")
+
+/obj/machinery/power/lgenerator/attackby(obj/item/W, mob/user)
+	if (istype(W, /obj/item/tank/))
+		if (src.internalTank)
+			user.show_text("There appears to be a tank loaded already.", "red")
+			return
+		if (src.check_tank(W) == 0)
+			user.show_text("The tank doesn't contain any plasma.", "red")
+			return
+		src.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>")
+		user.u_equip(W)
+		W.set_loc(src)
+		src.internalTank = W
+		src.UpdateIcon()
+		tgui_process.try_update_ui(user, src)
 
-		src.updateUsrDialog()
-		return
+	else if (istype(W, /obj/item/cell))
+		if (src.internalCell)
+			user.show_text("There appears to be a power cell inserted already.", "red")
+			return
+		src.visible_message("<span class='notice'>[user] loads [W] into the [src].</span>")
+		user.u_equip(W)
+		W.set_loc(src)
+		src.internalCell = W
+		tgui_process.try_update_ui(user, src)
 
-	update_icon()
-		if (src.active)
-			src.UpdateOverlays(spin_sprite, "spin")
-			light.enable()
-		else
-			src.UpdateOverlays(null, "spin")
-			light.disable()
+	else
+		..()
 
-		if (src.P)
-			tank_sprite.icon_state = "ggen-tank"
-			src.UpdateOverlays(tank_sprite, "tank")
-		else
-			src.UpdateOverlays(null, "tank")
+	return
+
+/obj/machinery/power/lgenerator/update_icon()
+	if (src.active)
+		src.UpdateOverlays(spin_sprite, "spin")
+		light.enable()
+	else
+		src.UpdateOverlays(null, "spin")
+		light.disable()
+
+	if (src.internalTank)
+		tank_sprite.icon_state = "ggen-tank"
+		src.UpdateOverlays(tank_sprite, "tank")
+	else
+		src.UpdateOverlays(null, "tank")
+
+	return
+
+/obj/machinery/power/lgenerator/proc/APC_check()
+	if (!src)
+		return 0
+
+	var/area/A = get_area(src)
+	if (!A || !A.requires_power)
+		return 0
+
+	var/obj/machinery/power/apc/AC = get_local_apc(src)
+	if (!AC)
+		return 0
+	if (AC && !AC.cell)
+		return 2
```

Comment:
Magic numbers aren't great here. Possible to use defined constants? These *seem* to be enum type things at a glance. Appreciate this is just moved code, no issue if you don't want to touch with this PR to avoid bloating it (something for someone to do as a bit of tidy up work afterwards, though!)